azure_functions_worker/bindings/shared_memory_manager.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from __future__ import annotations
+import uuid
+from typing import Dict
+from typing import Optional
+from ..logging import logger
+from ..mmap_handler.file_writer import FileWriter
+from ..mmap_handler.file_reader import FileReader
+from ..mmap_handler.file_accessor_factory import FileAccessorFactory
+
+
+class SharedMemoryManager:
+    """
+    Performs all operations related to reading/writing data from/to shared memory.
+    This is used for transferring input/output data of the function from/to the functions host over
+    shared memory as opposed to RPC to improve the rate of data transfer and the function's
+    end-to-end latency.
+    """
+    def __init__(self):
+        # The allocated memory maps are tracked here so that a reference to them is kept open until
+        # they have been used (e.g. if they contain a function's output, it is read by the
+        # functions host).
+        # Having a mapping of the name and the memory map is then later used to close a given
+        # memory map by its name, after it has been used.
+        # key: map_name, val: mmap.mmap
+        self.allocated_mem_maps: Dict[str, mmap.mmap] = {}
+        self.file_accessor = FileAccessorFactory.create_file_accessor()
+        self.file_reader = FileReader()
+        self.file_writer = FileWriter()
+
+    def is_enabled(self) -> bool:
+        """
+        Whether supported types should be transferred between functions host and the worker using
+        shared memory.
+        """
+        return True
```

Comment:
Do we want to check the app settings or whichever mechanism we follow in the C# side?

Reply:
Yes, I have to find out if they have some API for doing this in the Python side or I should just check the env variable.